src/views/ettersending/Ettersending.tsx

```diff
@@ -69,6 +69,13 @@ export function Ettersending(props: IProps) {
       krav.svar === DOKUMENTKRAV_SVAR_SENDER_IKKE
   );
 
+  const missingDokumentkrav: IDokumentkrav[] = availableDokumentkravForEttersending.filter(
```

Comment:
Istedet for å først hente ut availableDokumentkravForEttersending og så kjøre filter på den så kan vi gjøre alt i en operasjon:
```
const missingDokumentkrav = props.dokumentkrav.krav.filter(
    (krav) =>
      (krav.svar === DOKUMENTKRAV_SVAR_SEND_NAA || krav.svar === DOKUMENTKRAV_SVAR_SENDER_SENERE) &&
      !krav.bundleFilsti
  );


const receivedDokumentkrav = props.dokumentkrav.krav.filter(
    (krav) =>
      (krav.svar === DOKUMENTKRAV_SVAR_SEND_NAA || krav.svar === DOKUMENTKRAV_SVAR_SENDER_SENERE) &&
      krav.bundleFilsti
  );
```

Reply:
Kan dette være en løsning? En kombinasjon av det du nevnte og Hanne

```
  const availableDokumentkravForEttersending = props.dokumentkrav.krav;

  const missingDokumentkrav: IDokumentkrav[] = availableDokumentkravForEttersending.filter(
    (krav: IDokumentkrav) =>
      krav.svar === DOKUMENTKRAV_SVAR_SEND_NAA ||
      krav.svar === DOKUMENTKRAV_SVAR_SENDER_SENERE ||
      !krav.bundleFilsti
  );
  const receivedDokumentkrav: IDokumentkrav[] = availableDokumentkravForEttersending.filter(
    (krav: IDokumentkrav) =>
      krav.svar === DOKUMENTKRAV_SVAR_SEND_NAA ||
      krav.svar === DOKUMENTKRAV_SVAR_SENDER_SENERE ||
      krav.bundleFilsti
  );
  ```

Reply:
Jeg tenker litt på lesbarhet

Reply:
Ellers synes jeg dette er lesbart, man s er tilgjengelig dokumentkrav og så tydeliggjør man hvilke som er mottatt og mangler

```
  const availableDokumentkravForEttersending = props.dokumentkrav.krav.filter(
    (krav: IDokumentkrav) =>
      krav.svar === DOKUMENTKRAV_SVAR_SEND_NAA || krav.svar === DOKUMENTKRAV_SVAR_SENDER_SENERE
  );

  const missingDokumentkrav: IDokumentkrav[] = availableDokumentkravForEttersending.filter(
    (krav: IDokumentkrav) => !krav.bundleFilsti
  );
  const receivedDokumentkrav: IDokumentkrav[] = availableDokumentkravForEttersending.filter(
    (krav: IDokumentkrav) => krav.bundleFilsti
  );
  ```

Reply:
hmm, props.dokumentkrav.krav er absolutt alle dokumentkrav, og det er ikke alle som er tilgjengelig for ettersending så det blir missvisende. Hvis du vil bli kvitt props så kan du jo desctructure props så blir det kanskje litt mer ryddig?

Reply:
Tenker du typ

```
const { krav } = props.dokumentkrav;

  const availableDokumentkravForEttersending = krav.filter(
    (krav: IDokumentkrav) =>
      krav.svar === DOKUMENTKRAV_SVAR_SEND_NAA || krav.svar === DOKUMENTKRAV_SVAR_SENDER_SENERE
  );

  const missingDokumentkrav: IDokumentkrav[] = availableDokumentkravForEttersending.filter(
    (krav: IDokumentkrav) => !krav.bundleFilsti
  );
  const receivedDokumentkrav: IDokumentkrav[] = availableDokumentkravForEttersending.filter(
    (krav: IDokumentkrav) => krav.bundleFilsti
  );

  const unavailableDokumentkravForEttersending = krav.filter(
    (krav: IDokumentkrav) =>
      krav.svar === DOKUMENTKRAV_SVAR_SEND_NOEN_ANDRE ||
      krav.svar === DOKUMENTKRAV_SVAR_SENDT_TIDLIGERE ||
      krav.svar === DOKUMENTKRAV_SVAR_SENDER_IKKE
  );
```

Reply:
Tenkte noe sånt: 
```
export function Ettersending({ dokumentkrav }: IProps) {
  const missingDokumentkrav = dokumentkrav.krav.filter(
    (krav) =>
      (krav.svar === DOKUMENTKRAV_SVAR_SEND_NAA || krav.svar === DOKUMENTKRAV_SVAR_SENDER_SENERE) &&
      !krav.bundleFilsti
  );

  const receivedDokumentkrav = dokumentkrav.krav.filter(
    (krav) =>
      (krav.svar === DOKUMENTKRAV_SVAR_SEND_NAA || krav.svar === DOKUMENTKRAV_SVAR_SENDER_SENERE) &&
      krav.bundleFilsti
  );

  const unavailableDokumentkravForEttersending = dokumentkrav.krav.filter(
    (krav) =>
      krav.svar === DOKUMENTKRAV_SVAR_SEND_NOEN_ANDRE ||
      krav.svar === DOKUMENTKRAV_SVAR_SENDT_TIDLIGERE ||
      krav.svar === DOKUMENTKRAV_SVAR_SENDER_IKKE
  );

```

Reply:
Skjønner, fungerer dette som et kompromis? For feilsøking sin del, kan det være nyttig at `availableDokumentkravForEttersending` er sin egen ting siden den har felles filter for både `missingDokumentkrav` og `receivedDokumentkrav`

```
export function Ettersending({ dokumentkrav }: IProps) {
const availableDokumentkravForEttersending = dokumentkrav.krav.filter(
    (krav: IDokumentkrav) =>
      krav.svar === DOKUMENTKRAV_SVAR_SEND_NAA || krav.svar === DOKUMENTKRAV_SVAR_SENDER_SENERE
  );

  const missingDokumentkrav: IDokumentkrav[] = availableDokumentkravForEttersending.filter(
    (krav: IDokumentkrav) => !krav.bundleFilsti
  );
  const receivedDokumentkrav: IDokumentkrav[] = availableDokumentkravForEttersending.filter(
    (krav: IDokumentkrav) => krav.bundleFilsti
  );

  const unavailableDokumentkravForEttersending = dokumentkrav.krav.filter(
    (krav: IDokumentkrav) =>
      krav.svar === DOKUMENTKRAV_SVAR_SEND_NOEN_ANDRE ||
      krav.svar === DOKUMENTKRAV_SVAR_SENDT_TIDLIGERE ||
      krav.svar === DOKUMENTKRAV_SVAR_SENDER_IKKE
  );
  ```

Reply:
Kjør på 🚀